# 0 Installation

Please install the torch with specific PyTorch and CUDA version by following the instruction in the official website: https://pytorch.org/get-started/locally/


In [ ]:
%pip install torch==2.4.0 --index-url https://download.pytorch.org/whl/cu121

In [1]:
import torch
import os

In [2]:
print(f"PyTorch version: {torch.__version__}")
print(f"Is CUDA available: {torch.cuda.is_available()}")
print(f"CUDA version: {torch.version.cuda}")
print(f"Device count: {torch.cuda.device_count()}")
print(f"Current device: {torch.cuda.current_device()}")

PyTorch version: 2.4.0+cu121
Is CUDA available: True
CUDA version: 12.1
Device count: 1
Current device: 0


In [3]:
os.environ['TORCH'] = torch.__version__
print(torch.__version__)

2.4.0+cu121


# Install PyTorch Geometric

https://pytorch-geometric.readthedocs.io/en/latest/install/installation.html

In [ ]:
%pip install torch-geometric

## Additional Libraries

If you want to utilize the full set of features from PyG, there exists several additional libraries you may want to install:

- [pyg-lib](https://github.com/pyg-team/pyg-lib): Heterogeneous GNN operators and graph sampling routines
- [torch-scatter](https://github.com/rusty1s/pytorch_scatter): Accelerated and efficient sparse reductions
- [torch-sparse](https://github.com/rusty1s/pytorch_sparse): SparseTensor support, see here
- [torch-cluster](https://github.com/rusty1s/pytorch_cluster): Graph clustering routines
- [torch-spline-conv](https://github.com/rusty1s/pytorch_spline_conv): SplineConv support

These packages come with their own CPU and GPU kernel implementations based on the  [PyTorch C++/CUDA/hip(ROCm) extension interface](https://github.com/pytorch/extension-cpp/). For a basic usage of PyG, these dependencies are fully optional. We recommend to start with a minimal installation, and install additional dependencies once you start to actually need them.

## Installation from Wheels

For ease of installation of these extensions, we provide pip wheels for these packages for all major OS, PyTorch and CUDA combinations, see [here](https://data.pyg.org/whl/):

In [4]:
import torch

print(torch.__version__)    # check torch version
print(torch.version.cuda)   # check cuda version

2.4.0+cu121
12.1


Then install the packages via pip:

```bash
pip install pyg_lib torch_scatter torch_sparse torch_cluster torch_spline_conv -f https://data.pyg.org/whl/torch-${TORCH}+${CUDA}.html
```

![pyg](./src/pyg.png)

since our
- torch.__version__: 2.4.0
- torch.version.cuda: 12.1

We need to replace `${TORCH}` with `2.4.0` and `${CUDA}` with `cu121` in the above command.
    
```bash
pip install pyg_lib torch_scatter torch_sparse torch_cluster torch_spline_conv -f https://data.pyg.org/whl/torch-2.4.0+cu121.html
```

In [ ]:
%pip install pyg_lib torch_scatter torch_sparse torch_cluster torch_spline_conv -f https://data.pyg.org/whl/torch-2.4.0+cu121.html